Survey/SurveyUnitTests/Sources/Mock/Login/MockLoginRepository.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  MockLoginRepository.swift
+//  Survey
+//
+//  Created by Taher on 14/11/22.
+//
+
+import Combine
+import Foundation
+
+final class MockLoginRepository: LoginRepositoryProtocol {
+
+    var isLoginSuccess = false
+
+    func getLoginToken(email: String, password: String) -> Observable<LoginTokenApi> {
+        if email == "abc@nimblehq.co", password == "12345" {
+            isLoginSuccess = true
+        }
+        return Fail(error: HTTPError.invalidUrl)
+            .eraseToAnyPublisher()
+    }
+}
```

Comment:
You don't have to (and shouldn't) add logic in mock type 🙏. By logic I means check values of the inputs (`email`, `password`). Make a setter for returning value, then in the function just return it. Also the logic here looks a little strange to me since you set `isLoginSuccess` to `true` but still return a `Fail`.
```
final class MockLoginRepository: LoginRepositoryProtocol {

    var getLoginTokenReturnValue: Observable<LoginTokenApi>!

    func getLoginToken(email: String, password: String) -> Observable<LoginTokenApi> {
        return getLoginTokenReturnValue
    }
 }
```
When you test `LoginUseCase` (which will use this mocked repository), depend on the case you want to test (success or failure), use it accordingly:
```
mockRepository. getLoginTokenReturnValue = Just(dummyLoginTokenApi)
```
or:
```
mockRepository. getLoginTokenReturnValue = Fail(error: someError)
```